packages/core/src/runtime-configurations.ts

```diff
@@ -0,0 +1,138 @@
+import type { TopLevelConfig } from './index';
+
+declare global {
+    interface Window {
+        engineEntryOptions?: (options: { urlParams: URLSearchParams; envName: string }) => URLSearchParams;
+    }
+}
+
+export type ConfigModule = {
+    default: TopLevelConfig;
+};
+
+export type ConfigLoader = () => Promise<ConfigModule[]>;
+
+export type ConfigLoaders = Record<string, ConfigLoader>;
+
+export class RuntimeConfigurations {
```

Comment:
plz add comment explain what this class is used for an how... and read what gpt writes

Reply:
added comments to each public function and the class itself